pandas/core/ops/__init__.py

```diff
@@ -1041,80 +1050,60 @@ def wrapper(self, other, axis=None):
         elif isinstance(other, ABCSeries) and not self._indexed_same(other):
             raise ValueError("Can only compare identically-labeled Series objects")
 
-        elif is_categorical_dtype(self):
+        elif (
+            is_list_like(other)
+            and len(other) != len(self)
+            and not isinstance(other, frozenset)
+        ):
+            # TODO: why are we treating len-1 frozenset differently?
```

Comment:
Where is this coming from (a test?) I don't see this check in the existing code, it feels a bad idea to introduce such a check if we don't know for which reason

Reply:
a test, yes.  I dont know which off the top of my head, but this was needed to maintain existing behavior.  I'd like to change it to maintain/achieve consistency with do things elsewhere.

Reply:
Looks like the test that makes this necessary is in tests.series.test_operators.TestSeriesComparisons.test_comparison_tuples

```
s = Series([frozenset([1]), frozenset([1, 2])])
result = s == frozenset([1])
```

Supported listlikes in object-dtype is a hassle

Reply:
OK, so the frozenset was an example test case, but there might be others that previously worked like that. So I don't think the solution is to add an explicit frozenset check as done now. Do you know how it got through the code in the previous version?

Reply:
previously the length check was only done on ndarray/Index/Series

Reply:
> previously the length check was only done on ndarray/Index/Series

The let's go back to that logic? (or eg at least when the dtype is object?) As this will now break comparison of object dtypes that hold list-like scalar objects

Reply:
sure.  we should have a larger discussion on how we can handle this more consistently across the codebase